test/ui/cypress/e2e/deployment_driver.cy.js

```diff
@@ -18,38 +11,101 @@ describe('Deployment driver web UI', () => {
+    cy.get('p').contains('To manage or setup new subscription, visit the')
+    cy.get('p > .pf-c-button')
+      .contains('Red Hat Hybrid Cloud Console')
+      .should('have.attr', 'href', 'https://console.redhat.com/')
+
+    // Check the H1 title
+    cy.get('h1')
+      .contains('Ansible Automation Platform Deployment Engine')
+      .should('be.visible') 
   })
 
   it('left navigation has expected navigation links', () => {
+    cy.get('#nav-toggle').click()
     cy.get('div#page-sidebar').as('navigation')
+
     cy.get('@navigation').contains('Deployment')
+      .should('be.visible')
+      .should('have.attr','href').and('include','/')
+
     cy.get('@navigation').contains('Documentation')
+      .should('be.visible')
+      .should('have.attr','href').and('include', '/documentation')
+
     cy.get('@navigation').contains('Logout')
+      .should('be.visible')    
   })
 
   it('list of deployment steps contains expected steps', () => {
-    cy.get('div.deploy-step>ul').as('steps')
-    cy.get('@steps').contains('VNET and_Subnets')
-    cy.get('@steps').contains('Private DNS')
-    cy.get('@steps').contains('AAP Repository')
-    cy.get('@steps').contains('Database Server and Databases')
-    cy.get('@steps').contains('AKS Cluster')
-    cy.get('@steps').contains('AAP Operators')
-    cy.get('@steps').contains('AAP_Applications')
-    cy.get('@steps').contains('Application Ingress')
-    cy.get('@steps').contains('Seeded Content')
-    cy.get('@steps').contains('Billing')
+    // Check the title of Deployment Steps
+    cy.get('.pf-c-title')
+      .contains('Deployment Steps')
+      .should('be.visible')
+
+    // Check the title of Deployment Steps
+    cy.get('.pf-c-title')
+      .contains('Deployment Steps')
+      .should('be.visible')
+    
+    // Check the detailed deployment step's names in the Deployment Steps panel
+    cy.fixture('deployment_steps.json').then((deploymentStepsData) => {
+      let total_steps = deploymentStepsData.steps.length
+      cy.log('Total Deployment Steps: '+total_steps)
+      
+      cy.get('div.deploy-step>ul').as('steps')
+      for (let i=0; i<total_steps; i++) {
+        cy.log('Deployment Step '+ deploymentStepsData.steps[i].stepNo +": " + deploymentStepsData.steps[i].Name)
+        
+        cy.get('@steps')
+          .contains(`${deploymentStepsData.steps[`${i}`].Name}`)
+      }
+    })
+
   })
 
   it('clicking cancel button brings up a dialog that can be closed', () => {
+    // Check the Cancel Deployment button on the main screen
     cy.get('div.deployProgress').as('statusArea')
     cy.get('@statusArea').get('button.cancelButton').click()
+
+    // Check the UI elements on the pupped up dialog after clicking the Cancel Deployment button from the main screen
     cy.get('div[role="dialog"]').as('dialog')
-    cy.get('@dialog').contains('Cancel Deployment')
+    cy.get('@dialog', {timeout: 4000}).contains('Cancel Deployment')
```

Comment:
Why do we need a 4 second wait time here, cypress has built-in waits.

Reply:
During my testing, sometimes this step would fail with the default cypress timeout (3000). So I used `{timeout:4000}` for this step particularly.

Reply:
Interesting, I have not seen it but let's keep it to make the test more robust.